raiden/connection_manager.py

```diff
@@ -109,14 +111,29 @@ def connect(
                     funding
                 )
 
-    def leave(self, wait_for_settle=True, timeout=30):
+    def leave_async(self, wait_for_settle=True, timeout=240):
+        """ Async version of `leave()`
+        Args:
+            wait_for_settle (bool): block until successful settlement?
+            timeout (float): maximum time to wait in seconds
         """
-        Leave the token network.
+        leave_result = AsyncResult()
+        gevent.spawn(
+            self.leave,
+            wait_for_settle=wait_for_settle,
+            timeout=timeout,
+            async_result=leave_result
+        )
+        return leave_result
+
+    def leave(self, wait_for_settle=True, timeout=30, async_result=AsyncResult()):
```

Comment:
I clarified the scope of the timeout in the docstring: it's only for the waiting time for settlement. Not sure what you mean with

> be variable if we also want to `wait _for_settle`

?

> shouldn't it be at least as big as the biggest `settle_timeout` channel from the joined channels

I thought about it, but discarded that option. I believe the caller should take care of the timeout (in seconds, not blocks). Unfortunately, unless we have some measurement for block times, the best we can do is use a large-ass timeout.

For now I added a `TODO` to `raiden_service.py::leave_all_token_networks_async`.